ui/app/models/secret-engine.js

```diff
@@ -95,43 +95,57 @@ export default SecretEngineModel.extend({
   }),
 
   formFieldGroups: computed('engineType', function () {
+    let defaultFields = ['path'];
+    let optionFields;
     const type = this.engineType;
-    let defaultGroup;
-    // KV has specific config options it adds on the enable engine. https://www.vaultproject.io/api/secret/kv/kv-v2#configure-the-kv-engine
-    if (type === 'kv') {
-      defaultGroup = { default: ['path', 'maxVersions', 'casRequired', 'deleteVersionAfter'] };
-    } else {
-      defaultGroup = { default: ['path'] };
+    const CORE_OPTIONS = ['description', 'config.listingVisibility', 'local', 'sealWrap'];
+
+    switch (type) {
```

Comment:
* dust * -  could this just directly be `this.engineType`? would help readability a bit